Userland/Libraries/LibCrypto/Cipher/ChaCha20.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2022, stelar7 <dudedbz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+
+namespace Crypto::Cipher {
+
+class ChaCha20 {
+    static constexpr u32 CONSTANT_16_BYTES[] { 0x61707865, 0x3120646E, 0x79622D36, 0x6B206574 };
+    static constexpr u32 CONSTANT_32_BYTES[] { 0x61707865, 0x3320646E, 0x79622D32, 0x6B206574 };
+
+public:
+    ChaCha20(ReadonlyBytes key, ReadonlyBytes nonce, u32 initial_counter = 0);
+    ErrorOr<ByteBuffer> run_cipher(ReadonlyBytes input);
```

Comment:
Let's at least keep the interface somewhat consistent and add encrypt()/decrypt()
It'd also be nice if the main impl took a `Bytes output`, I believe the code in TLS expects something like that (have you checked if any TLS bits can be reused for this?)

Reply:
Havent looked at anything tls yet, thats step 3 :^)